src/app/3d/qgs3dmaptoolpaintbrush.cpp

```diff
@@ -0,0 +1,165 @@
+/***************************************************************************
+  qgs3dmaptoolpaintbrush.cpp
+  --------------------------------------
+  Date                 : January 2025
+  Copyright            : (C) 2025 by Matej Bagar
+  Email                : matej dot bagar at lutraconsulting dot co dot uk
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgs3dmaptoolpaintbrush.h"
+#include "moc_qgs3dmaptoolpaintbrush.cpp"
+#include "qgs3deditutils.h"
+#include "qgsrubberband3d.h"
+#include "qgs3dutils.h"
+#include "qgscameracontroller.h"
+#include "qgswindow3dengine.h"
+#include "qgsframegraph.h"
+#include "qgsgeometry.h"
+#include "qgsguiutils.h"
+#include "qgslinestring.h"
+
+
+class QgsPointCloudAttribute;
+Qgs3DMapToolPaintBrush::Qgs3DMapToolPaintBrush( Qgs3DMapCanvas *canvas )
+  : Qgs3DMapToolPointCloudChangeAttribute( canvas )
+{
+}
+
+Qgs3DMapToolPaintBrush::~Qgs3DMapToolPaintBrush() = default;
+
+void Qgs3DMapToolPaintBrush::run()
+{
+  QgsTemporaryCursorOverride busyCursor( Qt::WaitCursor );
+
+  const QgsGeometry searchSkeleton = QgsGeometry( new QgsLineString( mDragPositions ) );
+  const QgsGeometry searchPolygon = searchSkeleton.buffer( mSelectionRubberBand->width() / 2, 37 );
```

Comment:
why 37 vertices per quarter of a circle? :slightly_smiling_face:   It feels like too much - generally we want fewer vertices so that the intersection test does not take too long. Probably 4-8 should be enough